### ISSUE TITLE
Generated draft release not pubicly accessible

### ISSUE THREAD

**Issue by arjenderijke** [grafana/plugin-actions#14]
I use the [this](https://github.com/grafana/plugin-actions/blob/main/build-plugin/README.md) workflow to build a grafana plugin. The "softprops/action-gh-release@v1" action is used to build a github release. It is a draft release, because this is the first version of my plugin and i cannot sign the package yet. The release seems to be ok at first, because you can download the generated artifacts from the webinterface. But it turns out that these urls don't work when you try to download the artifacts from the commandline, for example with curl or wget. And this means that you cannot use these urls in the "create plugin submission" form on grafana cloud. Because if you use these urls in that submission form, it gives the same error message, status 404. 

I looked online and this does not seem to be a new problem. Before i used a older workflow that was generated by the "create-plugin" package. That had the same issue. I was hoping that this newer workflow would have solved the problem. The fact that you can download the artifacts from the webinterface suggest that the plugin is created, but that there is a problem with the permissions. Because the release page is also unavailable from a browser where you are not logged in.

**Comment by academo** [comment]
@arjenderijke have you tried to manually setting the draft releases to published releases? This is the default behaviour of github and we don't have any influence on that part. 

Something we can do is add a configuration field so the action generates a real release instead of a draft if passed. What do you think?

**Comment by arjenderijke** [comment]
Yes, i tried that later and that works. My plugin is published now. But the experience is a bit confusing, because this happens when you try to publish a plugin for the first time. At that point you don't know what to expect. The text that is generated for the release suggests that you can use the links for the submission of your plugin. Which is not true, because the it is a draft release. And you only notice this when the plugin submission generates and error. The error message itself is confusing, because as a logged in user, you have the required permissions to download the generated artifact. And it is even more confusing, because when you publish a plugin for the first time, you also have to deal with the fact that you cannot sign the plugin yet. How you have to deal with that problem is also a bit confusing.

Now that i know how it works, i think that the choice to generate a draft release is correct. Maybe there is reason why the release cannot be public yet. So i don't think it is a problem that the user has to manually set the release from draft to public. But the text of the release should be changed to mention this explicitly. And also mention that the links only work for a published release. Then it should be easier to figure out what is the problem, especially in the case when you try to publish a plugin for the first time.

As far as i remember, the tutorial about publishing plugins also does not mention this. So it would be helpful if that tutorial also mentions this issue and the solution. And for that specific tutorial it would also be an improvement if it would mention what to do when you publish your first plugin but don't have the permissions to sign it yet.

Once you know how it works, publishing a plugin is not difficult at all. But without the experience, it is easy to make the mistakes i made.

**Comment by academo** [comment]
Thanks for the feedback. [I created an issue](https://github.com/grafana/plugin-actions/issues/18) so we work in this. You are also most welcome to submit a PR to change this message if you wish. 

I also created a [second issue](https://github.com/grafana/plugin-tools/issues/802) in plugin-tools so we add a message in the packaging and sign pages indicating that first time submitters don't need to sign the plugin since it is not yet possible.
